_includes/_code/lakes-interactive/lib/border-star-button.dart

```diff
@@ -0,0 +1,153 @@
+import 'package:flutter/material.dart';
+// Uncomment lines 3 and 6 to view the visual layout at runtime.
+//import 'package:flutter/rendering.dart' show debugPaintSizeEnabled;
+
+void main() {
+  //debugPaintSizeEnabled = true;
+  runApp(new MyApp());
+}
+
+class FavoriteWidget extends StatefulWidget {
+  @override
+  FavoriteWidgetState createState() => new FavoriteWidgetState();
+}
+
+class FavoriteWidgetState extends State<FavoriteWidget> {
+  IconData currentIcon = Icons.star_border;
+  int favoriteCount = 40;
+
+  void toggleFavorite() {
+    setState(() {
+      // If the campground is currently favorited, unfavorite it.
+      if (currentIcon == Icons.star) {
+        favoriteCount = 40;
+        currentIcon = Icons.star_border;
+      // Otherwise, favorite it.
+      } else {
+        favoriteCount = 41;
+        currentIcon = Icons.star;
+      }
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Row(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        new IconButton(
+          icon: new Icon(currentIcon),
+          color: Colors.red[500],
+          onPressed: toggleFavorite,
+        ),
+        new Text(favoriteCount.toString()),
```

Comment:
In the actual version  (lakes-interactive/lib/main.dart) I put a fixed width on the Container, which worked. I will change it to a SizedBox.  (And this example will be deleted, as it was an intermediate version.)